### PR TITLE
v3.4.2-rhel: Fix intermittently missing files in merged directory during llistxattr

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
Addresses: https://bugzilla.redhat.com/show_bug.cgi?id=2008997 in the
v3.4.2-rhel branch.  Bumps c/storage to the version that was available
with this PR: https://github.com/containers/storage/pull/1037

Signed-off-by: tomsweeneyredhat <tsweeney@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@mheon @giuseppe PTAL.  This is my attempt to backport containers/storage#1037.  I did so by vendoring in the commit from that pr in c/storage (https://github.com/containers/storage/commit/7e016f3cfa9e869eb75c03e896dfe177ea80e69d), but the vendoring dragged in a whole lot more than I expected. Is this the right approach?

FWIW, we need to wrap this up by Monday in order to avoid the exception dance.

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by Luap99** [comment]
@TomSweeneyRedHat  I think you have to backport this commit in c/storage first to the 1.36 branch if the intention is to only get the changes from that one PR.

**Comment by TomSweeneyRedHat** [comment]
Hmm, the latest branch we have in c/storage is release-1.33.  @rhatdan, WDYT about creating a release-1.37 branch from the v1.37.0 release, and then I'll backport that change there, then backport it all here?

Or is it better to go with this as is?

I got a LGTM from Giuseppe and happy green test buttons, but I'm leary about the extra baggage this dragged in.

**Comment by rhatdan** [comment]
I am fine with creating that branch.

**Comment by rhatdan** [comment]
https://github.com/containers/storage/tree/release-1.37

**Comment by TomSweeneyRedHat** [comment]
Ugh, as I was merging into Podman, I realized I need c/storage v1.36 NOT v1.37.

**Comment by TomSweeneyRedHat** [comment]
Repushed using the right branch in c/storage, v1.36.  There is a lot less files here, but it did drag in some json related files that I was not expecting.  From c/storage, we're now down to just the expected changes in store.go to cover the issue.

File: go.mod
Comment (vrothberg) [inline go.mod:21]: Why are we bumping to a commit and not to a specific tag/version?

Comment (TomSweeneyRedHat) [reply go.mod:21]: It's a bug fix that I was trying to get through quickly on a weekend.  I probably should have created a new release.  @jnovy what is the absolute drop dead time for this?

Comment (vrothberg) [reply go.mod:21]: I fear that the quick-and-dirty could cause confusion in the future. I am OK to merge it as is as long as we cut a release right after and bump it here.

**Comment by umohnani8** [comment]
Let's merge and we can bump it to an actual release after as Valentin suggested? @TomSweeneyRedHat @vrothberg please remove the hold if you agree

/lgtm
/hold

**Comment by TomSweeneyRedHat** [comment]
@vrothberg FYI, just spun a PR in c/storage to bump the version. Once done, I'll tag and then do another version dance here.

https://github.com/containers/storage/pull/1100

**Comment by TomSweeneyRedHat** [comment]
I'm in GIT heck in my sandbox.  I'm going to kill this PR and will create a new PR using the new v1.36.1 release

**Comment by TomSweeneyRedHat** [comment]
Replacement PR: https://github.com/containers/podman/pull/12789

**Pull request closed without merging**
